src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -336,6 +336,7 @@ private static NamedTypeSymbol GetTupleUnderlyingType(ImmutableArray<TypeSymbol>
             if ((object)diagnostics != null && (object)syntax != null)
             {
                 Binder.ReportUseSiteDiagnostics(firstTupleType, diagnostics, syntax);
+                Binder.ReportDiagnosticsIfObsoleteInternal(diagnostics, firstTupleType, syntax, firstTupleType.ContainingType, BinderFlags.None);
```

Comment:
The pair of methods is called in four places and the pair share the same parameters. And use-site errors and obsolete errors seem similar enough to group here.